### PR TITLE
Make a few doc sections more explicit

### PR DESCRIPTION
**Pull request by kenahoo**
In general, if a Description section jumped right into exceptions or special cases without first saying what the function does, I added something briefly describing what the function does.  

The rest of what's here is mostly small clarifications or corrections.

### CODE REVIEW

**Comment by kenahoo** [comment]
Are the Travis & Appveyor failures expected?  The errors look unrelated to the changes in the PR.

File: R/manip.r [hunk 1]
```diff
@@ -66,16 +66,18 @@ filter_ <- function(.data, ..., .dots = list(), .preserve = TRUE) {
...
 #'
-#' Positive values select rows to keep; negative values drop rows. The
-#' values provided must be either all positive or all negative.
-#'
 #' @family single table verbs
 #' @param .data A tbl.
-#' @param ... Integer row values.
+#' @param ... Integer row values.  Provide either positive values to keep,
+#'   or negative values to drop. The values provided must be either all
+#'   positive or all negative.
```
Comment (hadley) [inline R/manip.r]: Mention what happens if the row doesn't exist?

File: R/manip.r [hunk 2]
```diff
@@ -66,16 +66,18 @@ filter_ <- function(.data, ..., .dots = list(), .preserve = TRUE) {
 
 #' Select rows by position
```
Comment (hadley) [inline R/manip.r]: Can we use something other than select here?

File: R/manip.r [hunk 3]
```diff
@@ -116,10 +119,14 @@ slice_ <- function(.data, ..., .dots = list()) {
   UseMethod("slice_")
 }
 
-#' Reduces multiple values down to a single value
+#' Reduce multiple values down to a single value
+#'
+#' Create one or more scalar variables summarizing the variables of an existing tbl.
+#'
+#' When `summarise()` is used on grouped data created by [group_by()],
```
Comment (hadley) [inline R/manip.r]: I think this needs to go in the description. Should also describe what happens when not grouped.

File: R/manip.r [hunk 4]
```diff
@@ -199,16 +206,14 @@ summarize <- summarise
 summarize_ <- summarise_
 
 
-#' Add new variables
+#' Create or transform variables
 #'
-#' `mutate()` adds new variables and preserves existing;
-#' `transmute()` drops existing variables.
+#' `mutate()` adds new variables and preserves existing ones;
```
Comment (hadley) [inline R/manip.r]: Mention that both preserve the number of rows?

File: R/manip.r [hunk 5]
```diff
@@ -199,16 +206,14 @@ summarize <- summarise
...
+#' `transmute()` adds new variables and drops existing ones.
 #'
-#' @section Useful functions:
-#'
-#' * [`+`], [`-`] etc
+#' @section Useful functions available in calculations of variables:
 #'
-#' * [log()]
+#' * [`+`], [`-`], [log()], etc., for their usual arithmetic meanings
```
Comment (hadley) [inline R/manip.r]: Arithmetic -> mathematical?

File: R/manip.r [hunk 6]
```diff
@@ -233,7 +238,7 @@ summarize_ <- summarise_
 #' @inheritParams filter
 #' @inheritSection filter Tidy data
 #' @param ... Name-value pairs of expressions. Use `NULL` to drop
-#'   a variable.
+#'   a variable in `mutate`.
```
Comment (hadley) [inline R/manip.r]: Mention (before `NULL`) that each expression must yield a result of length 1 or the same length as the group?

File: R/manip.r [hunk 7]
```diff
@@ -344,13 +349,14 @@ transmute_.grouped_df <- function(.data, ..., .dots = list()) {
...
-#' @param ... Comma separated list of unquoted variable names. Use
-#'   [desc()] to sort a variable in descending order.
+#' @param ... Comma separated list of unquoted variable names, or expressions
+#'   involving variable names. Use [desc()] to sort a variable in descending order.
 #' @family single table verbs
 #' @return An object of the same class as `.data`.
 #' @examples
 #' arrange(mtcars, cyl, disp)
 #' arrange(mtcars, desc(disp))
+#' arrange(mtcars, (disp-150)^2)
```
Comment (hadley) [inline R/manip.r]: I don't think arranging by computed variables is useful enough to include in an example.

File: R/manip.r [hunk 8]
```diff
@@ -386,12 +392,13 @@ arrange.grouped_df <- function(.data, ..., .by_group = FALSE) {
 
 #' Select/rename variables by name
 #'
+#' Choose variables from a tbl, optionally renaming them.
```
Comment (hadley) [inline R/manip.r]: That's only `select()`, not `rename()` (which you can't use to drop variables)

File: R/manip.r [hunk 9]
```diff
@@ -386,12 +392,13 @@ arrange.grouped_df <- function(.data, ..., .by_group = FALSE) {
...
 #' `select()` keeps only the variables you mention; `rename()`
 #' keeps all variables.
 #'
 #' @section Useful functions:
 #' As well as using existing functions like `:` and `c()`, there are
-#' a number of special functions that only work inside `select`
+#' a number of special functions that only work inside `select`:
```
Comment (hadley) [inline R/manip.r]: `select` -> `select()`

File: R/manip.r [hunk 10]
```diff
@@ -416,17 +423,18 @@ arrange.grouped_df <- function(.data, ..., .by_group = FALSE) {
...
-#'   Positive values select variables; negative values to drop variables.
+#'   Positive values select variables; negative values drop variables.
 #'   If the first expression is negative, `select()` will automatically
 #'   start with all variables.
 #'
-#'   Use named arguments to rename selected variables.
+#'   Use named arguments, e.g. `new_name=old_name`, to rename selected variables.
 #'
-#'   These arguments are automatically [quoted][rlang::quo] and
+#'   The arguments in `...` are automatically [quoted][rlang::quo] and
```
Comment (hadley) [inline R/manip.r]: Should be the same in all these functions

File: R/manip.r [hunk 11]
```diff
@@ -416,17 +423,18 @@ arrange.grouped_df <- function(.data, ..., .by_group = FALSE) {
...
+#'   You can treat variable names like they are positions, so you can
+#'   use expressions like `x:y` to select ranges of variables.
 #'
-#'   Positive values select variables; negative values to drop variables.
+#'   Positive values select variables; negative values drop variables.
 #'   If the first expression is negative, `select()` will automatically
 #'   start with all variables.
 #'
-#'   Use named arguments to rename selected variables.
+#'   Use named arguments, e.g. `new_name=old_name`, to rename selected variables.
```
Comment (hadley) [inline R/manip.r]: Needs space around `=`

File: R/manip.r (hunk 2 above)
Comment (kenahoo) [reply R/manip.r]: Indeed - I used "choose" elsewhere to avoid ambiguity, will do the same here.

File: R/manip.r (hunk 4 above)
Comment (kenahoo) [reply R/manip.r]: Will do.

File: R/manip.r (hunk 3 above)
Comment (kenahoo) [reply R/manip.r]: Will do - please check my wording to make sure it's correct.

File: R/manip.r (hunk 5 above)
Comment (kenahoo) [reply R/manip.r]: Sure.

File: R/manip.r (hunk 1 above)
Comment (kenahoo) [reply R/manip.r]: Wasn't sure whether that behavior was set yet - looks like it silently ignores indices outside the range of data, is that what we should document?

File: R/manip.r [hunk 12]
```diff
@@ -66,16 +66,18 @@ filter_ <- function(.data, ..., .dots = list(), .preserve = TRUE) {
 
 #' Select rows by position
 #'
+#' Select rows by their ordinal position in the tbl.  Grouped tbls use
```
[the remark this answers is not in the document]
Comment (kenahoo) [reply R/manip.r]: I'm not sure how consistent the existing docs are - there's precedent for "tbl" in the docs for `distinct`, `do`, `explain`, `group_by`, etc.  There are also other terms used, e.g. "tibble" in `group_data` (which I think is just trying to be nonspecific for the various table types it accepts, but might be specifically a `tbl_df`?).  `nest_by` uses both in the same sentence.

File: R/manip.r (hunk 7 above)
Comment (kenahoo) [reply R/manip.r]: I do it fairly often - adding an example was my sneaky way to make sure it didn't get accidentally removed in a future revision. =)

File: R/manip.r (hunk 9 above)
Comment (kenahoo) [reply R/manip.r]: Will do.

File: R/manip.r (hunk 11 above)
Comment (kenahoo) [reply R/manip.r]: Got it.

File: R/manip.r (hunk 10 above)
Comment (kenahoo) [reply R/manip.r]: Will do.

File: R/manip.r (hunk 8 above)
Comment (kenahoo) [reply R/manip.r]: I'll rework that.

I also noticed that the RHS of `select()` expressions can't usefully be computed expressions right now:

```
> data.frame(x=1:5, y=10) %>% select(z=x)
  z
1 1
2 2
3 3
4 4
5 5
> data.frame(x=1:5, y=10) %>% select(z=2)
   z
1 10
2 10
3 10
4 10
5 10
> data.frame(x=1:5, y=10) %>% select(z=x+1)
  z10 z9
1   1 10
2   2 10
3   3 10
4   4 10
5   5 10
```

I think I would have expected it to either add 1 to the values of `x` (but that's what `mutate`/`transmute` are for), or select the column one position to the right of `x`, but it does neither.

I could add the following to the `Details` section if desired, I'll include it in a patch and you can consider it:

```
#' These functions work by column index, not value; thus, an expression
#' like `select(data.frame(x = 1:5, y = 10), z = x+1)` does not create a variable
#' with values `2:6`. (In the current implementation, the expression `z = x+1`
#' wouldn't do anything useful.)  To calculate using column values, see
#' [mutate()]/[transmute()].
```

Comment (hadley) [reply R/manip.r]: @lionel- do we have some text somewhere (maybe in a blog post) that describes this more precisely?

Comment (lionel-) [reply R/manip.r]: There is some information at https://dplyr.tidyverse.org/articles/dplyr.html#patterns-of-operations in the "Selecting operations" subsection.

File: R/manip.r (hunk 7 above)
Comment (krlmlr) [reply R/manip.r]: @hadley: Do we keep this behavior that we allow arranging by arbitrary expressions, or do we want to remove it at some point? The documentation should reflect this; if we keep, we should have a test for it too -- so far I only see `-` and `desc()` in the tests.

Comment (hadley) [reply R/manip.r]: We want to keep it, we just don't want to show it in the examples.

**Comment by krlmlr** [comment]
Thanks! I'll remove the example that arranges by an expression for now.

**Pull request merged**